sdk/eventhubs/azure_messaging_eventhubs/src/common/connection_manager.rs

```diff
@@ -417,17 +483,24 @@ impl ConnectionManager {
         }
 
         debug!("Performing authorization for {url}");
-        let session = AmqpSession::new();
-        session.begin(connection.as_ref(), None).await?;
-        let cbs = AmqpClaimsBasedSecurity::new(&session)?;
-        cbs.attach().await?;
-
-        let expires_at = new_token.expires_on;
-        cbs.authorize_path(
-            url.to_string(),
-            None,
-            new_token.token.secret().to_string(),
-            expires_at,
+
+        retry_azure_operation(
+            || async {
+                let connection = connection.clone();
+                let path = url.to_string();
```

Comment:
Could probably move this out of the loop. Shouldn't change, right? Anything else you could avoid unnecessarily reallocing?

Reply:
If I move it out of the loop, it fails because a `Fn` closure cannot capture local variables.  I tried changing the `Fn` to `FnMut` but the failure still happens.